assets/wizards/engagement/views/newsletters/index.js

```diff
@@ -120,20 +124,115 @@ export const NewspackNewsletters = ( { className, onUpdate, isOnboarding = true
 	);
 };
 
-const Newsletters = () => {
-	const [ { status, url, error, newslettersConfig }, updateConfiguration ] = hooks.useObjectState(
-		{}
+export const SubscriptionLists = ( { onUpdate } ) => {
+	const [ error, setError ] = useState();
+	const [ inFlight, setInFlight ] = useState( false );
+	const [ lists, setLists ] = useState( [] );
+	const updateConfig = data => {
+		setLists( data );
+		if ( typeof onUpdate === 'function' ) {
+			onUpdate( data );
+		}
+	};
+	const fetchLists = () => {
+		setInFlight( true );
+		apiFetch( {
+			path: '/newspack-newsletters/v1/lists',
+		} )
+			.then( updateConfig )
+			.catch( setError )
+			.finally( () => setInFlight( false ) );
+	};
+	const saveLists = () => {
+		setInFlight( true );
+		apiFetch( {
+			path: '/newspack-newsletters/v1/lists',
+			method: 'post',
+			data: { lists },
+		} )
+			.then( updateConfig )
+			.catch( setError )
+			.finally( () => setInFlight( false ) );
+	};
+	const handleChange = ( index, name ) => value => {
+		const newLists = [ ...lists ];
+		newLists[ index ][ name ] = value;
+		updateConfig( newLists );
+	};
+	useEffect( fetchLists, [] );
+	if ( ! lists?.length ) {
+		return null;
+	}
+	return (
+		<>
+			<SectionHeader
+				title={ __( 'Subscription Lists', 'newspack' ) }
+				description={ __( 'Manage the lists available for subscription.', 'newspack' ) }
+			/>
+			{ error && (
+				<Notice noticeText={ error.message || __( 'Something went wrong.', 'newspack' ) } isError />
+			) }
+			{ lists.map( ( list, i ) => (
```

Comment:
Suggestion for readability:

```suggestion
			{ lists.map( ( list, index ) => (
```

Reply:
Done in 290c959dec053dd26be8c9d50cf55a2af87e7177